### PR TITLE
Micu/feature #173 provide filter option for venue sub category

### PR DESCRIPTION
**Pull request by micuintus**

### CODE REVIEW

**Comment by jmastr** [comment]
this.patch.is.awesome!

Thank you very much. This is cool. The UX is pretty nice! I also like, that got rid of the "unknown" entries in the shopping list.

One remark: please do also not show the "Venue features" in the filter view, when "Shopping" is selected. Since all entries are "unknown", selecting one filter leads to an empty list.

**Comment by micuintus** [comment]
You're so damn right, good catch! Changed it.

**Comment by micuintus** [comment]
Oh, I just noticed that #177 should probably be catched with this PR?

**Comment by jmastr** [comment]
Thanks for 5bae503. Works perfectly. I found another bug, that I did not notice the first time: If one selects a venue feature and then switches to Shopping, the list is empty. In the background, the flags are probably still evaluated.

Proposal: Save the current flag state, set everything to the initial state before the switch to Shopping, restore the state when switching back to Food.

**Comment by jmastr** [comment]
Another bug, maybe n my part of the source code, is: switching from favorites view to venues list shows food and shopping entries regardless, what's selected in the filter view.

File: qml/components-generic/VenueDescriptionAlgorithms.js [hunk 1]
```diff
@@ -70,9 +70,13 @@ function restaurantCategory(key)
 
 function seatProperty(key)
 {
+    if (typeof key === "undefined")
+    {             //% "unknown"
+        return qsTrId("id-unknown");
+    }
+
     switch (key)
-    {
```
Comment (jmastr) [inline qml/components-generic/VenueDescriptionAlgorithms.js]: diff is not necessary

**Comment by jmastr** [comment]
impressive!

**Pull request merged**